packages/abi-coder/src/interface.ts

```diff
@@ -150,6 +150,29 @@ export default class Interface {
     return concat([selector, new BooleanCoder().encode(isRef), args]);
   }
 
+  encodeMainFunctionData(
```

Comment:
These two methods are almost identical; do we need them both?

 - `encodeMainFunctionData`
 - `encodeFunctionData`

I see the little differences in the code, but I need help understanding them.

I guess the questions are:
 1. Are `contract` functions different from `script` functions?
 2. Could something else be changed _(in the `ScriptFactory` or elsewhere)_ to avoid duplicating this method?